internal/controller/kyma_controller.go

```diff
@@ -129,6 +129,15 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	// Suppress the creation of further events based on an "unauthorized against the SKR cluster" condition
+	// to not put burst loads onto ETCD when doing credential rotation.
+	// This is observed to be the first place where such condition appears in the Kyma reconcile loop.
+	if util.IsUnauthorized(err) {
+		r.deleteRemoteClientCache(ctx, kyma)
+		r.Metrics.RecordRequeueReason(metrics.KymaUnauthorized, metrics.UnexpectedRequeue)
+		return ctrl.Result{Requeue: true}, nil
```

Comment:
I didn't do it because this would mean we are recording an event in [`requeueWithError->updateStatusWithError->`](https://github.com/kyma-project/lifecycle-manager/blob/main/internal/controller/kyma_controller.go#L569) which we try to avoid. But right, I missed that the status needs to be updated. Will think about how to best achieve that without writing the event.

Reply:
Addressed this now by a minor refactoring. See commit [0fb198f](https://github.com/kyma-project/lifecycle-manager/pull/1275/commits/0fb198f8d3e3c9d7c1f4e76b187f8e8d10de7aeb) which moves the writing of the event from `updateStatusWithError` to `requeueWithError`. The behavior should be the same as before.

But now, I can call `updateStatusWithError` without writing the event, see [d15f716](https://github.com/kyma-project/lifecycle-manager/pull/1275/commits/d15f716192191061a0bdb9dc4c6f39200320b5fc)

The resulting status of the KymaCR is now:

```yaml
apiVersion: operator.kyma-project.io/v1beta2
kind: Kyma
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"operator.kyma-project.io/v1beta2","kind":"Kyma","metadata":{"annotations":{"skr-domain":"example.domain.com"},"name":"kyma-sample","namespace":"kcp-system"},"spec":{"channel":"regular"}}
    kyma-kcp-crd-generation: "2"
    kyma-skr-crd-generation: "1"
    moduletemplate-kcp-crd-generation: "2"
    moduletemplate-skr-crd-generation: "1"
    skr-domain: example.domain.com
  creationTimestamp: "2024-02-06T12:13:29Z"
  finalizers:
  - operator.kyma-project.io/Kyma
  generation: 1
  labels:
    operator.kyma-project.io/managed-by: lifecycle-manager
  name: kyma-sample
  namespace: kcp-system
  resourceVersion: "5377"
  uid: 574e9225-fd22-4eb6-882f-a96f2b8cf36e
spec:
  channel: regular
status:
  activeChannel: regular
  conditions:
  - lastTransitionTime: "2024-02-06T12:49:12Z"
    message: modules state is unknown
    observedGeneration: 1
    reason: Ready
    status: Unknown
    type: Modules
  - lastTransitionTime: "2024-02-06T12:49:12Z"
    message: module templates synchronization state is unknown
    observedGeneration: 1
    reason: Ready
    status: Unknown
    type: ModuleCatalog
  - lastTransitionTime: "2024-02-06T12:49:12Z"
    message: skrwebhook is out of sync and needs to be resynchronized
    observedGeneration: 1
    reason: Ready
    status: Unknown
    type: SKRWebhook
  lastOperation:
    lastUpdateTime: "2024-02-06T12:49:12Z"
    operation: 'initializing sync context failed: failed to ensure remote namespace
      exists: failed to get API group resources: unable to retrieve the complete list
      of server APIs: v1: Unauthorized'
  modules:
  - channel: regular
    fqdn: kyma-project.io/module/template-operator
    manifest:
      apiVersion: operator.kyma-project.io/v1beta2
      kind: Manifest
      metadata:
        generation: 1
        name: kyma-sample-template-operator-2235966007
        namespace: kcp-system
    name: template-operator
    resource:
      apiVersion: operator.kyma-project.io/v1alpha1
      kind: Sample
      metadata:
        name: sample-yaml
        namespace: kyma-system
    state: Ready
    template:
      apiVersion: operator.kyma-project.io/v1beta2
      kind: ModuleTemplate
      metadata:
        generation: 1
        name: template-operator-regular
        namespace: kcp-system
    version: 0.1.0
  state: Error
```

Behavior of events, logs and metrics is still as described in PR description